app/controllers/users_controller.rb

```diff
@@ -1,5 +1,25 @@
 class UsersController < ApplicationController
   def show
-    @user = User.find_by(username: params[:username])
+      @user = User.find_by(username: params[:username])
+      redirect_to posts_path, alert: "Could not find a user with that name" unless @user
+  end
+
+  def follow
+    other_user = params[:user]
```

Comment:
Hmm. So, `other_user` here isn't a `User`, but rather an id. I see that in `is_following?` you're doing the `User.find`. I think it'd be better to convert it to a `User` here.